In [1]:
import numpy as np
import pandas as pd
import matplotlib as plt

In [2]:
df = pd.read_csv(r'C:\Users\rafae\OneDrive\Desktop\hackton\teste\forma.csv', engine='python')

In [3]:
df.head()

FORMA_ATUACAO  %_EM_RELACAO_AO_BRASIL  \
0    Estabelecimento fixo,"2,374","39,7675","336566"                     NaN   
1  Em local fixo, fora da loja,"0,5606","9,3906",...                     NaN   
2  Porta a Porta, postos móveis ou por ambulantes...                     NaN   
3      Máquinas automáticas,"0,0459","0,7683","6502"                     NaN   
4               Internet,"0,7178","12,0236","101760"                     NaN   

   %_EM_RELACAO_A_UF  No._MEI  
0                NaN      NaN  
1                NaN      NaN  
2                NaN      NaN  
3                NaN      NaN  
4                NaN      NaN

In [38]:
dg = pd.read_html('https://www.ibge.gov.br/indicadores#desemprego')

COMPRAS ECOMMERCE

1) Encontre a quantidade de linhas.

In [5]:
ecom = pd.read_csv('C:\\Users\\rafae\\Python-Data-Science-and-Machine-Learning-Bootcamp\\2. Python para análise de dados\\Pandas\\Pandas Exercises\\Ecommerce Purchases')

In [6]:
ecom.head()

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                   Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3              7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4  23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4  Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   
2       Simpson, Williams and Pham      675957666125       08/19   
3  Williams, Marshall and Buchanan  6011578504430710       02/24   
4        Brown, Watson and Andrews  6011456623207998       10/25   

   CC Security Code                  CC Provider  \
0               900                 JCB 16 digit   
1               561                   Mastercard   
2               699                 JCB 16 digit   
3               384                     Discover   
4               678  Diners Club / Carte Blanche   

                            Email                                     Job  \
0               pdunlap@yahoo.com  Scientist, product/process development   
1              anthony41@reed.com                       Drilling engineer   
2  amymiller@morales-harrison.com                Customer service manager   
3     brent16@olson-robinson.info                       Drilling engineer   
4     christopherwright@gmail.com                             Fine artist   

        IP Address Language  Purchase Price  
0  149.146.147.205       el           98.14  
1     15.160.41.51       fr           70.73  
2   132.207.160.22       de            0.95  
3     30.250.74.19       es           78.04  
4     24.140.33.94       es           77.82

In [7]:
ecom.shape

(10000, 14)

Resposta: teremos um total de 10.000 linhas

2)Qual o preço de compra médio

In [8]:
ecom['Purchase Price'].mean()

50.34730200000025

In [9]:
ecom.mean()

Credit Card         2.341374e+15
CC Security Code    9.072178e+02
Purchase Price      5.034730e+01
dtype: float64

RESPOSTA = A média será de 50,34. Pode ser encontrada atraves do método específico ecom['Purchase Price'].mean() ou de maneira mais geral com o comando: ecom.mean()

3)Qual o preço de compra mais alto e mais baixo

In [10]:
ecom['Purchase Price'].max()

99.99

In [11]:
ecom['Purchase Price'].min()

0.0

RESPOSTA: Teremos como valor máximo 99.99 e valor mínimo 0.0

4)Quantas pessoas têm Inglês 'en' como sua língua de escolha no site?

In [12]:
ecom[(ecom['Language']=='en')].count()

Address             1098
Lot                 1098
AM or PM            1098
Browser Info        1098
Company             1098
Credit Card         1098
CC Exp Date         1098
CC Security Code    1098
CC Provider         1098
Email               1098
Job                 1098
IP Address          1098
Language            1098
Purchase Price      1098
dtype: int64

In [13]:
ecom[(ecom['Language']=='en')].shape

(1098, 14)

5)QUANTAS PESSOAS TEM O CARGO DE ADVOGADO

In [14]:
ecom[ecom['Job']=='Lawyer'].count()

Address             30
Lot                 30
AM or PM            30
Browser Info        30
Company             30
Credit Card         30
CC Exp Date         30
CC Security Code    30
CC Provider         30
Email               30
Job                 30
IP Address          30
Language            30
Purchase Price      30
dtype: int64

In [15]:
ecom[ecom['Job']=='Lawyer'].shape

(30, 14)

6) QUANTAS PESSOAS FIZERAM A COMPRA DURANTE AM E DURANTE PM.

In [16]:
ecom[(ecom['AM or PM']=='AM')].count()

Address             4932
Lot                 4932
AM or PM            4932
Browser Info        4932
Company             4932
Credit Card         4932
CC Exp Date         4932
CC Security Code    4932
CC Provider         4932
Email               4932
Job                 4932
IP Address          4932
Language            4932
Purchase Price      4932
dtype: int64

In [17]:
ecom[(ecom['AM or PM']=='PM')].count()

Address             5068
Lot                 5068
AM or PM            5068
Browser Info        5068
Company             5068
Credit Card         5068
CC Exp Date         5068
CC Security Code    5068
CC Provider         5068
Email               5068
Job                 5068
IP Address          5068
Language            5068
Purchase Price      5068
dtype: int64

RESPOSTA ALTERNATIVA:

In [18]:
ecom['AM or PM'].value_counts()

PM    5068
AM    4932
Name: AM or PM, dtype: int64

7) Quais são os títulos de trabalho mais comum.

In [19]:
ecom['Job'].value_counts()

Interior and spatial designer    31
Lawyer                           30
Social researcher                28
Designer, jewellery              27
Purchasing manager               27
                                 ..
Stage manager                     7
Actuary                           6
Editor, film/video                6
Information systems manager       5
Investment analyst                5
Name: Job, Length: 623, dtype: int64

RESPOSTA ALTERNATIVA:

In [55]:
ecom['Job'].value_counts().head()

Interior and spatial designer    31
Lawyer                           30
Social researcher                28
Designer, jewellery              27
Purchasing manager               27
Name: Job, dtype: int64

RESPOSTA: Os mais comuns são:
         Interio and spatial designer,
         Lawyer,
         Social researcher,
         Designer, jewellery,
         Purchasing manager,
        

8) Alguém fez uma compra que veio do Lot: "90 WT", qual foi o preço de compra para esta transação?

In [40]:
ecom[ecom['Lot']=='90 WT']['Purchase Price']

513    75.1
Name: Purchase Price, dtype: float64

RESPOSTA: A resposta será de 75.1

9) Qual é o e-mail da pessoa com o seguinte número do cartão de crédito: 4926535242672853 

In [41]:
ecom[ecom['Credit Card']==4926535242672853]['Email']

1234    bondellen@williams-garza.com
Name: Email, dtype: object

RESPOSTA: A resposta será "bondellen@williams-garza.com"

10) Quantas pessoas têm o American Express como seu fornecedor de cartão de crédito * e * fizeram uma compra acima de US $ 95?

In [42]:
ecom['CC Provider']

0                      JCB 16 digit
1                        Mastercard
2                      JCB 16 digit
3                          Discover
4       Diners Club / Carte Blanche
                   ...             
9995                   JCB 15 digit
9996                   JCB 16 digit
9997                  VISA 16 digit
9998               American Express
9999                   JCB 15 digit
Name: CC Provider, Length: 10000, dtype: object

In [43]:
ecom[(ecom['CC Provider']=='American Express') & (ecom['Purchase Price']> 95)].count()

Address             39
Lot                 39
AM or PM            39
Browser Info        39
Company             39
Credit Card         39
CC Exp Date         39
CC Security Code    39
CC Provider         39
Email               39
Job                 39
IP Address          39
Language            39
Purchase Price      39
dtype: int64

Resposta Alternativa

In [44]:
ecom[(ecom['CC Provider']=='American Express') & (ecom['Purchase Price']> 95)].shape

(39, 14)

RESPOSTA: A resposta será de 39 pessoas

Será utilizado o código: ecom[(ecom['CC Provider']=='American Express') & (ecom[Purchase Price']>95)].shape

10) Quantas pessoas tem um cartão de crédito que expira em 2025?

In [45]:
sum(ecom['CC Exp Date'].apply(lambda x: x[3:]) == '25')

1033

RESPOSTA: A resposta será 1033 pessoas tem o vencimento do cartão em 2025. 
O termo '[3]' é referente a 'partir de qual string', será feita a análise.

11) Quais são os 5 principais provedores de e-mail / hosts mais populares (por exemplo, gmail.com, yahoo.com, etc ...)

In [46]:
ecom['Email'].apply(lambda x: x.split('@')[1]).value_counts().head(5)

hotmail.com     1638
yahoo.com       1616
gmail.com       1605
smith.com         42
williams.com      37
Name: Email, dtype: int64